error when trying to import cookies from file

when i tried to open a cookies.txt file in yt-dlp-compose, i got this error:

<img width="374" alt="Screenshot 2024-06-07 at 5 35 21 PM" src="https://github.com/StefanLobbenmeier/yt-dlp-compose/assets/86898409/e0cd9fc7-dbb3-4a45-ba25-4ffc6feaf6af">

**steps to reproduce:**
1. open yt-dlp-compose
2. open settings
3. scroll down to "cookies from file"
4. click the page icon
5. see error

**app version:** snapshot 33
**OS version:** macOS 14.5

Oh good find. I was able to reproduce and this is the full stacktrace:

```
Exception in thread "AWT-EventQueue-0" java.lang.NoClassDefFoundError: sun/misc/Unsafe
	at org.lwjgl.system.MemoryUtil.getUnsafeInstance(MemoryUtil.java:2948)
	at org.lwjgl.system.MemoryUtil.<clinit>(MemoryUtil.java:112)
	at org.lwjgl.system.Pointer$Default.<clinit>(Pointer.java:67)
	at com.darkrockstudios.libraries.mpfilepicker.FileChooserKt.chooseFile(FileChooser.kt:11)
	at com.darkrockstudios.libraries.mpfilepicker.DesktopFilePickerKt$FilePicker$1.invokeSuspend(DesktopFilePicker.kt:31)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:104)
	at androidx.compose.ui.platform.FlushCoroutineDispatcher$flush$1.invoke(FlushCoroutineDispatcher.skiko.kt:90)
	at androidx.compose.ui.platform.FlushCoroutineDispatcher$flush$1.invoke(FlushCoroutineDispatcher.skiko.kt:78)
	at androidx.compose.ui.platform.FlushCoroutineDispatcher.performRun(FlushCoroutineDispatcher.skiko.kt:99)
	at androidx.compose.ui.platform.FlushCoroutineDispatcher.flush(FlushCoroutineDispatcher.skiko.kt:78)
	at androidx.compose.ui.scene.ComposeSceneRecomposer.performScheduledEffects(ComposeSceneRecomposer.skiko.kt:88)
	at androidx.compose.ui.scene.BaseComposeScene.render(BaseComposeScene.skiko.kt:165)
	at androidx.compose.ui.scene.ComposeSceneMediator$DesktopSkikoView.onRender(ComposeSceneMediator.desktop.kt:523)
	at org.jetbrains.skiko.SkiaLayer.update$skiko(SkiaLayer.awt.kt:548)
	at org.jetbrains.skiko.redrawer.AWTRedrawer.update(AWTRedrawer.kt:54)
	at org.jetbrains.skiko.redrawer.MetalRedrawer$frameDispatcher$1.invokeSuspend(MetalRedrawer.kt:82)
	at org.jetbrains.skiko.redrawer.MetalRedrawer$frameDispatcher$1.invoke(MetalRedrawer.kt)
	at org.jetbrains.skiko.redrawer.MetalRedrawer$frameDispatcher$1.invoke(MetalRedrawer.kt)
	at org.jetbrains.skiko.FrameDispatcher$job$1.invokeSuspend(FrameDispatcher.kt:33)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:104)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(Unknown Source)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(Unknown Source)
	at java.desktop/java.awt.EventQueue$4.run(Unknown Source)
	at java.desktop/java.awt.EventQueue$4.run(Unknown Source)
	at java.base/java.security.AccessController.doPrivileged(Unknown Source)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(Unknown Source)
	at java.desktop/java.awt.EventQueue.dispatchEvent(Unknown Source)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(Unknown Source)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(Unknown Source)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(Unknown Source)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(Unknown Source)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(Unknown Source)
	at java.desktop/java.awt.EventDispatchThread.run(Unknown Source)
	Suppressed: kotlinx.coroutines.internal.DiagnosticCoroutineContextException: [androidx.compose.ui.scene.ComposeContainer$DesktopCoroutineExceptionHandler@5fd3e574, androidx.compose.runtime.BroadcastFrameClock@3491aa3b, StandaloneCoroutine{Cancelling}@853c518, FlushCoroutineDispatcher@24cdaeae]
Caused by: java.lang.ClassNotFoundException: sun.misc.Unsafe
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(Unknown Source)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(Unknown Source)
	at java.base/java.lang.ClassLoader.loadClass(Unknown Source)
	... 35 more
```

--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -71,6 +71,9 @@ compose {
                     TargetFormat.Exe,
                     TargetFormat.Msi,
                 )
+
+                // lwjgl3 needs sun.misc.Unsafe to be included in the bundled JRE
+                modules("jdk.unsupported")
             }
         }
     }

